gms/gossiper.cc

```diff
@@ -2013,7 +2013,6 @@ future<> gossiper::add_local_application_state(std::list<std::pair<application_s
             if (!es) {
                 auto err = format("endpoint_state_map does not contain endpoint = {}, application_states = {}",
                                   ep_addr, states);
-                logger.error("{}", err);
```

Comment:
Looks good to me - it's definitely not a good idea to log the same error twice. My only question is whether the appropriate error levels for such errors is "error" or "warning". When does this error happen? Does it indicate some serious logic bug or a trivial case of unresponsive node?

Reply:
I don't think it is a serious bug. The only window when it can happen (according to the code) is when the gossiper is not yet started and someone tries to set local application state - a.k.a gossip some information about itself. Even if this doesn't succeed it is not that bad since when starting the gossiper, we build our entire local state, so succeeding this function is only critical for app state updates. Another indication of this issue "minorness" is the fact that the exception that is thrown and then caught a few lines below just in order to log the warning and nothing else.

Reply:
Ohh, and it is not unresponsive not - it is the local state so we are talking about our own state which only exists when gossip is running.